fs/remote/blob.go

```diff
@@ -263,82 +241,35 @@ func (b *blob) fetchRegions(allData map[region]io.Writer, fetched map[region]boo
 	b.lastCheck = time.Now()
 	b.lastCheckMu.Unlock()
 
-	// chunk responsed data. Regions must be aligned by chunk size.
-	// TODO: Reorganize remoteData to make it be aligned by chunk size
 	for {
-		reg, p, err := mr.Next()
+		_, p, err := mr.Next()
 		if err == io.EOF {
 			break
 		} else if err != nil {
 			return errors.Wrapf(err, "failed to read multipart resp")
 		}
-		if err := b.walkChunks(reg, func(chunk region) (retErr error) {
-			w := io.Discard
-
-			// If this chunk is one of the targets, write the content to the
-			// passed reader too.
-			if _, ok := fetched[chunk]; ok {
-				w = io.MultiWriter(w, allData[chunk])
-			}
-
-			// Copy the target chunk
-			if _, err := io.CopyN(w, p, chunk.size()); err != nil {
-				return err
-			}
-
-			b.fetchedRegionSetMu.Lock()
-			b.fetchedRegionSet.add(chunk)
-			b.fetchedRegionSetMu.Unlock()
-			fetched[chunk] = true
-			return nil
-		}); err != nil {
-			return errors.Wrapf(err, "failed to get chunks")
-		}
-	}
 
-	// Check all chunks are fetched
-	var unfetched []region
-	for c, b := range fetched {
-		if !b {
-			unfetched = append(unfetched, c)
+		if _, err := io.CopyN(w, p, reg.size()); err != nil {
+			return err
 		}
-	}
-	if unfetched != nil {
-		return fmt.Errorf("failed to fetch region %v", unfetched)
-	}
 
-	return nil
-}
+		b.fetchedRegionSetMu.Lock()
+		b.fetchedRegionSet.add(reg)
+		b.fetchedRegionSetMu.Unlock()
+		fetched = true
+	}
 
-// fetchRange fetches all specified chunks from remote blob.
-func (b *blob) fetchRange(allData map[region]io.Writer, opts *options) error {
-	if len(allData) == 0 {
-		return nil
+	if !fetched {
+		return fmt.Errorf("failed to fetch region %v", reg)
 	}
 
-	fetched := make(map[region]bool)
-	return b.fetchRegions(allData, fetched, opts)
+	return nil
 }
 
-type walkFunc func(reg region) error
-
-// walkChunks walks chunks from begin to end in order in the specified region.
-// specified region must be aligned by chunk size.
-func (b *blob) walkChunks(allRegion region, walkFn walkFunc) error {
-	if allRegion.b%b.chunkSize != 0 {
-		return fmt.Errorf("region (%d, %d) must be aligned by chunk size",
-			allRegion.b, allRegion.e)
-	}
-	for i := allRegion.b; i <= allRegion.e && i < b.size; i += b.chunkSize {
-		reg := region{i, i + b.chunkSize - 1}
-		if reg.e >= b.size {
-			reg.e = b.size - 1
-		}
-		if err := walkFn(reg); err != nil {
-			return err
-		}
-	}
-	return nil
+// fetchRange fetches content from remote blob.
+func (b *blob) fetchRange(reg region, w io.Writer, opts *options) error {
+	var fetched bool
```

Comment:
can you please just pass `false` here, since there's no use of `fetched` in the current context?

Reply:
Done